### PR TITLE
BIGTOP-3882: add support for ambari

### PR DESCRIPTION
**Pull request by MacChen01**
### Description of PR
Add support for ambari component, include compile command and problem.

### How was this patch tested?
build command         
           docker run --rm -v `pwd`:/home/bigtop --workdir /home/bigtop bigtop/slaves:trunk-openeuler-22.03-aarch64 bash -c '. /etc/profile.d/bigtop.sh; ./gradlew ambari-pkg'

smoke test command
           ./docker-hadoop.sh -C config_openeuler-22.03.yaml -c 3 -s

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (BIGTOP-3882)
- [ ] Make sure that newly added files do not have any licensing issues. When in doubt refer to https://www.apache.org/licenses/

### CODE REVIEW

**Comment by JiaLiangC** [comment]
@MacChen01 Could you please also submit a PR in the Apache Ambari community? The Ambari community is planning to support openEuler in version 2.9 release, and create a subtask under this issue https://issues.apache.org/jira/browse/AMBARI-25951

**Comment by MacChen01** [comment]
> @MacChen01 Could you please also submit a PR in the Apache Ambari community? The Ambari community is planning to support openEuler in version 2.9 release, and create a subtask under this issue https://issues.apache.org/jira/browse/AMBARI-25951

ok, it's my pleasure to support openEuler on Apache Ambari community.
I created a subtask for it:
https://issues.apache.org/jira/browse/AMBARI-25979
@JiaLiangC

**Comment by MacChen01** [comment]
**The smoke test successful log on openEuler:**
image

File: bigtop-packages/src/common/ambari/patch6-support-ppc.diff [hunk 1]
```diff
@@ -42,6 +42,14 @@ index 0634988657..022037b552 100644
 +          16,
 +          18
 +        ]
++      },
```
Comment (guyuqi) [inline bigtop-packages/src/common/ambari/patch6-support-ppc.diff:45]: This patch was for Powerpc.
It seems openEuler does not support ppc.

Comment (MacChen01) [reply bigtop-packages/src/common/ambari/patch6-support-ppc.diff]: About support for openEuler has moved from patch6-support-ppc.diff to patch14-support-openeuler.diff. @guyuqi 

code commit: https://github.com/apache/bigtop/pull/1147/commits/4c4db4a2d38e333ae3d15fca951136e9d3cdfdad

**Comment by MacChen01** [comment]
The smoke test successful log on openEuler:
image

**Comment by MacChen01** [comment]
**The smoke test successful log on fedora:**
image
https://gitee.com/macchen1/bigtop/blob/master/smoke-test-log/fedora-36-smoke-log/ambari-fedora.log

**The smoke test successful log on ubuntu:**
image

https://gitee.com/macchen1/bigtop/tree/master/smoke-test-log/ubuntu-22-smoke-log

File: bigtop-packages/src/common/ambari/patch14-support-openeuler.diff [hunk 2]
```diff
@@ -0,0 +1,97 @@
+diff -Naur apache-ambari-2.7.5-src/ambari-common/src/main/python/ambari_commons/os_check.py apache-ambari-2.7.5-src-openEuler/ambari-common/src/main/python/ambari_commons/os_check.py
```
Comment (guyuqi) [inline bigtop-packages/src/common/ambari/patch14-support-openeuler.diff:1]: This patch file would be applied to the other Distros.
But the other Distros don't need this patch file.

File: bigtop-deploy/puppet/modules/ambari/manifests/init.pp [hunk 3]
```diff
@@ -23,6 +23,39 @@
   }
 
   class server {
+    #install python2 and init postgresql for openEuler
+    if $operatingsystem == 'openEuler' {
+      exec { "download_python2.7":
```
Comment (guyuqi) [inline bigtop-deploy/puppet/modules/ambari/manifests/init.pp:28]: Python 2.7 would be installed in Bigtop_toolchain: https://github.com/apache/bigtop/blob/openEuler-support/bigtop_toolchain/manifests/packages.pp#L395

Please remove the python-2.7 installation from here.

**Review by guyuqi — CHANGES_REQUESTED** [review]

File: bigtop-deploy/puppet/modules/ambari/manifests/init.pp (hunk 3 above)
Comment (uscxiexing) [reply bigtop-deploy/puppet/modules/ambari/manifests/init.pp:28]: > Python 2.7 would be installed in Bigtop_toolchain: https://github.com/apache/bigtop/blob/openEuler-support/bigtop_toolchain/manifests/packages.pp#L395
> 
> Please remove the python-2.7 installation from here.

1，Python 2.7 is alrady installed in Bigtop_toolchain. 
2.   The openeuler was not support python2,and can not find in rpm-package-management.so we remove python2 require in openeuler. 
image
3.  when we use bigtop/puppet to run smoketest,can not find python2 in openeuler,and we can not to exec ambari-start or  ambari-agent. so we need to instadlled python in openeuler environment,and should add  this python code inbigtop-deploy/puppet/modules/ambari/manifests/init.pp
@guyuqi

File: bigtop-packages/src/common/ambari/patch14-support-openeuler.diff (hunk 2 above)
Comment (uscxiexing) [reply bigtop-packages/src/common/ambari/patch14-support-openeuler.diff:1]: > This patch file would be applied to the other Distros. But the other Distros don't need this patch file.

yes, ambari is not support openeuler, this patch is we add openuler code to supoort ambari.
1.  we run smoketest for other Distros  ,it not unaffected.
2.  if you want not add this patch in bigtop-packages/src/common/ambari/ ,Where should we place?
     add sed command in do-component-build ?
@guyuqi

Comment (guyuqi) [reply bigtop-packages/src/common/ambari/patch14-support-openeuler.diff:1]: @uscxiexing
Thanks for your efforts and elaboration.

>>if you want not add this patch in bigtop-packages/src/common/ambari/ ,Where should we place?
add sed command in do-component-build ?

Yes, it's a little bit tricky. 
If you kindly tested it on other Distros(Centos,Debian,Ubuntu ,Fedora, Rockylinux), 
we could temporarily put this patch file here (`bigtop-packages/src/common/ambari`).
I think we could remove this patch file after you post this patch in Ambari in the future.

File: bigtop-deploy/puppet/modules/ambari/manifests/init.pp (hunk 3 above)
Comment (guyuqi) [reply bigtop-deploy/puppet/modules/ambari/manifests/init.pp:28]: I think python is the fundamental tools.
What about moving python2 installation into `bigtop/puppet` for openEuler?

Comment (MacChen01) [reply bigtop-deploy/puppet/modules/ambari/manifests/init.pp]: > I think python is a fundamental tool. What about moving python2 installation into `bigtop/puppet` for openEuler?

The installation of python2 has moved into `bigtop/puppet`, the code commit:
f43bfb82391971fc0255b54f68c11da3ed59bf3a

@guyuqi

Comment (MacChen01) [reply bigtop-deploy/puppet/modules/ambari/manifests/init.pp]: Ambari smoke test successlly using bigtop/puppet, the log is:
image

The detail log is:
https://gitee.com/macchen1/bigtop/blob/master/smoke-test-log/openeuler-puppet-smoke-log/ambari-openeuler.log

@guyuqi 
We use the bigtop/puppet to smoke test is ok, please review the last PRs for openEuler.

**BIGTOP-3988: using bigtop/puppet for hadoop:**
https://github.com/apache/bigtop/pull/1168
**BIGTOP-3824: add support for ranger**
https://github.com/apache/bigtop/pull/1143
**BIGTOP-3967: add support for spark**
https://github.com/apache/bigtop/pull/1148

We can verfiy all components if you check and merge the PRs into openEuler-support branch, and we can fix the problem if it happen basing the branch.
Thanks.

**Comment by guyuqi** [comment]
Hi @MacChen01 @uscxiexing,
Sorry for the late reply. 

https://github.com/apache/bigtop/pull/1148, https://github.com/apache/bigtop/pull/1147 and https://github.com/apache/bigtop/pull/1143 have been verified on my local environment including building, toolchian images and smoke tests **separately**.

But the `postgresql-jdbc` service was deployed twice in [Ambari](https://github.com/apache/bigtop/pull/1147/files#diff-18357b51b0dfec2b4dafd4579f8d2469ef258984231733b80dbfc8aae5b93c02R28) and [Ranger](https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/modules/ranger/manifests/init.pp#L35).

Running the smoke tests of Ranger and Ambari together would cause them to fail:
```
./docker-hadoop.sh -d --create 3 --image bigtop/puppet:bgtp-openeuler-22.03 \
   --memory 16g --repo file:///bigtop-home/output --disable-gpg-check --enable-local-repo \
   --stack bigtop-utils,hdfs,yarn,spark,ambari,ranger \
   --smoke-tests spark,ambari,ranger
```

```
Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation 
Error: Error while evaluating a Resource Statement, Duplicate declaration: 
Package[postgresql-jdbc] is already declared at (file: /bigtop-home/bigtop-deploy/puppet/modules/ambari/manifests/init.pp, line: 28); 
cannot redeclare (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35) 
(file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35, column: 5) 
(file: /bigtop-home/bigtop-deploy/puppet/manifests/cluster.pp, line: 208) on node dc86e902b9ac.bigtop.apache.org
```

**Comment by MacChen01** [comment]
> Hi @MacChen01 @uscxiexing, Sorry for the late reply.
> 
> #1148, #1147 and #1143 have been verified on my local environment including building, toolchian images and smoke tests **separately**.
> 
> But the `postgresql-jdbc` service was deployed twice in [Ambari](https://github.com/apache/bigtop/pull/1147/files#diff-18357b51b0dfec2b4dafd4579f8d2469ef258984231733b80dbfc8aae5b93c02R28) and [Ranger](https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/modules/ranger/manifests/init.pp#L35).
> 
> Running the smoke tests of Ranger and Ambari together would cause them to fail:
> 
> ```
> ./docker-hadoop.sh -d --create 3 --image bigtop/puppet:bgtp-openeuler-22.03 \
>    --memory 16g --repo file:///bigtop-home/output --disable-gpg-check --enable-local-repo \
>    --stack bigtop-utils,hdfs,yarn,spark,ambari,ranger \
>    --smoke-tests spark,ambari,ranger
> ```
> 
> ```
> Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation 
> Error: Error while evaluating a Resource Statement, Duplicate declaration: 
> Package[postgresql-jdbc] is already declared at (file: /bigtop-home/bigtop-deploy/puppet/modules/ambari/manifests/init.pp, line: 28); 
> cannot redeclare (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35) 
> (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35, column: 5) 
> (file: /bigtop-home/bigtop-deploy/puppet/manifests/cluster.pp, line: 208) on node dc86e902b9ac.bigtop.apache.org
> ```



> postgresql



> Hi @MacChen01 @uscxiexing, Sorry for the late reply.
> 
> #1148, #1147 and #1143 have been verified on my local environment including building, toolchian images and smoke tests **separately**.
> 
> But the `postgresql-jdbc` service was deployed twice in [Ambari](https://github.com/apache/bigtop/pull/1147/files#diff-18357b51b0dfec2b4dafd4579f8d2469ef258984231733b80dbfc8aae5b93c02R28) and [Ranger](https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/modules/ranger/manifests/init.pp#L35).
> 
> Running the smoke tests of Ranger and Ambari together would cause them to fail:
> 
> ```
> ./docker-hadoop.sh -d --create 3 --image bigtop/puppet:bgtp-openeuler-22.03 \
>    --memory 16g --repo file:///bigtop-home/output --disable-gpg-check --enable-local-repo \
>    --stack bigtop-utils,hdfs,yarn,spark,ambari,ranger \
>    --smoke-tests spark,ambari,ranger
> ```
> 
> ```
> Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation 
> Error: Error while evaluating a Resource Statement, Duplicate declaration: 
> Package[postgresql-jdbc] is already declared at (file: /bigtop-home/bigtop-deploy/puppet/modules/ambari/manifests/init.pp, line: 28); 
> cannot redeclare (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35) 
> (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35, column: 5) 
> (file: /bigtop-home/bigtop-deploy/puppet/manifests/cluster.pp, line: 208) on node dc86e902b9ac.bigtop.apache.org
> ```

@guyuqi 
Can you get us some suggestions for it?
We installed the **_postgresql_** as the Ambari and Ranger need it for smoke test, where should we install it?

**Comment by guyuqi** [comment]
> > Hi @MacChen01 @uscxiexing, Sorry for the late reply.
> > #1148, #1147 and #1143 have been verified on my local environment including building, toolchian images and smoke tests **separately**.
> > But the `postgresql-jdbc` service was deployed twice in [Ambari](https://github.com/apache/bigtop/pull/1147/files#diff-18357b51b0dfec2b4dafd4579f8d2469ef258984231733b80dbfc8aae5b93c02R28) and [Ranger](https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/modules/ranger/manifests/init.pp#L35).
> > Running the smoke tests of Ranger and Ambari together would cause them to fail:
> > ```
> > ./docker-hadoop.sh -d --create 3 --image bigtop/puppet:bgtp-openeuler-22.03 \
> >    --memory 16g --repo file:///bigtop-home/output --disable-gpg-check --enable-local-repo \
> >    --stack bigtop-utils,hdfs,yarn,spark,ambari,ranger \
> >    --smoke-tests spark,ambari,ranger
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation 
> > Error: Error while evaluating a Resource Statement, Duplicate declaration: 
> > Package[postgresql-jdbc] is already declared at (file: /bigtop-home/bigtop-deploy/puppet/modules/ambari/manifests/init.pp, line: 28); 
> > cannot redeclare (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35) 
> > (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35, column: 5) 
> > (file: /bigtop-home/bigtop-deploy/puppet/manifests/cluster.pp, line: 208) on node dc86e902b9ac.bigtop.apache.org
> > ```
> 
> > postgresql
> 
> > Hi @MacChen01 @uscxiexing, Sorry for the late reply.
> > #1148, #1147 and #1143 have been verified on my local environment including building, toolchian images and smoke tests **separately**.
> > But the `postgresql-jdbc` service was deployed twice in [Ambari](https://github.com/apache/bigtop/pull/1147/files#diff-18357b51b0dfec2b4dafd4579f8d2469ef258984231733b80dbfc8aae5b93c02R28) and [Ranger](https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/modules/ranger/manifests/init.pp#L35).
> > Running the smoke tests of Ranger and Ambari together would cause them to fail:
> > ```
> > ./docker-hadoop.sh -d --create 3 --image bigtop/puppet:bgtp-openeuler-22.03 \
> >    --memory 16g --repo file:///bigtop-home/output --disable-gpg-check --enable-local-repo \
> >    --stack bigtop-utils,hdfs,yarn,spark,ambari,ranger \
> >    --smoke-tests spark,ambari,ranger
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation 
> > Error: Error while evaluating a Resource Statement, Duplicate declaration: 
> > Package[postgresql-jdbc] is already declared at (file: /bigtop-home/bigtop-deploy/puppet/modules/ambari/manifests/init.pp, line: 28); 
> > cannot redeclare (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35) 
> > (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35, column: 5) 
> > (file: /bigtop-home/bigtop-deploy/puppet/manifests/cluster.pp, line: 208) on node dc86e902b9ac.bigtop.apache.org
> > ```
> 
> @guyuqi Can you get us some suggestions for it? We installed the **_postgresql_** as the Ambari and Ranger need it for smoke test, where should we install it?


The Ambari smoke tests have been passed in almost all distributions in Bigtop 3.2.1.
`https://ci.bigtop.apache.org/job/Bigtop-3.2.1-smoke-tests/`


It seems postgreSQL was integrated into Ambari package.
May I ask why we need additionally add postgresql installation in Ambari? @MacChen01

**Comment by uscxiexing** [comment]
> 



> 
@guyuqi 
when not inited postgreSQL,I got a problem.Ambari-server setup error and Ambari failed to start psql by default..the log is :
image.
so i add init postgreSQL.

**Comment by uscxiexing** [comment]
> > > Hi @MacChen01 @uscxiexing, Sorry for the late reply.
> > > #1148, #1147 and #1143 have been verified on my local environment including building, toolchian images and smoke tests **separately**.
> > > But the `postgresql-jdbc` service was deployed twice in [Ambari](https://github.com/apache/bigtop/pull/1147/files#diff-18357b51b0dfec2b4dafd4579f8d2469ef258984231733b80dbfc8aae5b93c02R28) and [Ranger](https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/modules/ranger/manifests/init.pp#L35).
> > > Running the smoke tests of Ranger and Ambari together would cause them to fail:
> > > ```
> > > ./docker-hadoop.sh -d --create 3 --image bigtop/puppet:bgtp-openeuler-22.03 \
> > >    --memory 16g --repo file:///bigtop-home/output --disable-gpg-check --enable-local-repo \
> > >    --stack bigtop-utils,hdfs,yarn,spark,ambari,ranger \
> > >    --smoke-tests spark,ambari,ranger
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > ```
> > > Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation 
> > > Error: Error while evaluating a Resource Statement, Duplicate declaration: 
> > > Package[postgresql-jdbc] is already declared at (file: /bigtop-home/bigtop-deploy/puppet/modules/ambari/manifests/init.pp, line: 28); 
> > > cannot redeclare (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35) 
> > > (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35, column: 5) 
> > > (file: /bigtop-home/bigtop-deploy/puppet/manifests/cluster.pp, line: 208) on node dc86e902b9ac.bigtop.apache.org
> > > ```
> > 
> > 
> > > postgresql
> > 
> > 
> > > Hi @MacChen01 @uscxiexing, Sorry for the late reply.
> > > #1148, #1147 and #1143 have been verified on my local environment including building, toolchian images and smoke tests **separately**.
> > > But the `postgresql-jdbc` service was deployed twice in [Ambari](https://github.com/apache/bigtop/pull/1147/files#diff-18357b51b0dfec2b4dafd4579f8d2469ef258984231733b80dbfc8aae5b93c02R28) and [Ranger](https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/modules/ranger/manifests/init.pp#L35).
> > > Running the smoke tests of Ranger and Ambari together would cause them to fail:
> > > ```
> > > ./docker-hadoop.sh -d --create 3 --image bigtop/puppet:bgtp-openeuler-22.03 \
> > >    --memory 16g --repo file:///bigtop-home/output --disable-gpg-check --enable-local-repo \
> > >    --stack bigtop-utils,hdfs,yarn,spark,ambari,ranger \
> > >    --smoke-tests spark,ambari,ranger
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > ```
> > > Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation 
> > > Error: Error while evaluating a Resource Statement, Duplicate declaration: 
> > > Package[postgresql-jdbc] is already declared at (file: /bigtop-home/bigtop-deploy/puppet/modules/ambari/manifests/init.pp, line: 28); 
> > > cannot redeclare (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35) 
> > > (file: /bigtop-home/bigtop-deploy/puppet/modules/ranger/manifests/init.pp, line: 35, column: 5) 
> > > (file: /bigtop-home/bigtop-deploy/puppet/manifests/cluster.pp, line: 208) on node dc86e902b9ac.bigtop.apache.org
> > > ```
> > 
> > 
> > @guyuqi Can you get us some suggestions for it? We installed the **_postgresql_** as the Ambari and Ranger need it for smoke test, where should we install it?
> 
> The Ambari smoke tests have been passed in almost all distributions in Bigtop 3.2.1. `https://ci.bigtop.apache.org/job/Bigtop-3.2.1-smoke-tests/`
> 
> It seems postgreSQL was integrated into Ambari package. May I ask why we need additionally add postgresql installation in Ambari? @MacChen01

@guyuqi  
Thank you for verifying.
1.  when Individually verified ambari,if i not add init pgsql and got a error.The log is:
image
2.  The error like pgsql service is not start,when I add the command "/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data" to init pgsql in openeuler,the smoketest is ok. And other os did not needs.
3. I search some repea ambari setup pgsql source code .I use comand  "/usr/bin/postgresql-setup --init" in fedorea ,it run healthy.
image
but when i use this comand in openeuler, i got a error,the log is 
image
so i use the command "/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data"  to init pgsql in order to start pgsql.
4.  I notice you run-in ambari and ranger.but start ambari and ranger somketest in fedora ,i got a error, pgsql init  repea,the log is :
image

should we add this function to start ambari and ranger?

**Comment by guyuqi** [comment]
Thanks for your efforts and elaboration on this work. @uscxiexing 

>>The error like pgsql service is not start,when I add the command "/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data" to init pgsql >>in openeuler,the smoketest is ok. And other os did not needs.

You mean that Ambari could not start in openEuler because postgreSQL was not be started
by command ` /usr/bin/postgresql-setup --init`, and postgreSQL could be started by the other command: 
`/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data` in openEuler, right?   Please correct me if I have a misunderstanding.

IMO, what about removing postgre installation from puppet script and just adding your `initdb` command for openEuler?



>>I notice you run-in ambari and ranger.but start ambari and ranger somketest in fedora ,i got a error, pgsql init repea,the >>log is

If we change the `initdb` command to `/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data` and it could fix the issues when we run  Ranger and Ambari smoke tests together, just do it.

**Comment by MacChen01** [comment]
> Thanks for your efforts and elaboration on this work. @uscxiexing
> 
> > > The error like pgsql service is not start,when I add the command "/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data" to init pgsql >>in openeuler,the smoketest is ok. And other os did not needs.
> 
> You mean that Ambari could not start in openEuler because postgreSQL was not be started by command ` /usr/bin/postgresql-setup --init`, and postgreSQL could be started by the other command: `/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data` in openEuler, right? Please correct me if I have a misunderstanding.
> 
> IMO, what about removing postgre installation from puppet script and just adding your `initdb` command for openEuler?

1. yes, using the default init pgsql would got the error log:
image

so @uscxiexing using _**/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data**_ instead of it.
It's ok to verfiy ambari and ranger separately by this way in openEuler.

2. Do the smoke tests of ranger and ambari need to be run together? 
    We got the double init pgsql problem, if ambari and ranger smoke test together, the same problem happened in Fedora.
image


@guyuqi

**Comment by MacChen01** [comment]
> > > Do the smoke tests of ranger and ambari need to be run together?
> > > We got the double init pgsql problem, if ambari and ranger smoke test together, the same problem happened in Fedora.
> 
> It seems there are some conflicts exist in puppet deployment when postgre would be started twice (deploying Ranger and Ambari simultaneously).
> 
> I'm +1 for your solution. And please kindly add comments in this Ambari puppet script to explain why we change the command of `initdb` here and remind users that it woud fail to deploy Ranger and Ambari simultaneously. (We may fix it in the future)
> 
> Thanks a lot. @MacChen01 @uscxiexing

Delete the installation of postgerSQL and setup, the code is:
1d531b7d19da5a394b34df9a45785bab8590c783
Add the comment to remind users don't deploy Ranger and Ambari simultaneously: BIGTOP-3883.
@guyuqi

**Comment by uscxiexing** [comment]
> 
> Delete the installation of postgerSQL and setup, the code is: [1d531b7](https://github.com/apache/bigtop/commit/1d531b7d19da5a394b34df9a45785bab8590c783) Add the comment to remind users don't deploy Ranger and Ambari simultaneously: [BIGTOP-3883](https://issues.apache.org/jira/browse/BIGTOP-3883). @guyuqi

@MacChen01 @guyuqi 
Logs of successful modification：
image
image
the detail log：
https://gitee.com/macchen1/bigtop/blob/master/smoke-test-log/aarch64/openeuler-22.03-smoke-log/ambari-openeuler-22.03.log

**Comment by MacChen01** [comment]
We found the reason that postgreSQL didn't start, becasue the dbus service didn't install and start in openEuler puppet default.
So using the default Init postgreSQL mode, The code is:
cd6cd0d62a4a648cdb6d2186af4f722a960d0ee3

@guyuqi please check the code, thanks.

**Comment by guyuqi** [comment]
> We found the reason that postgreSQL didn't start, becasue the dbus service didn't install and start in openEuler puppet default. So using the default Init postgreSQL mode, The code is: [cd6cd0d](https://github.com/apache/bigtop/commit/cd6cd0d62a4a648cdb6d2186af4f722a960d0ee3)
> 
> @guyuqi please check the code, thanks.

Thanks for your continuing efforts to improve this PR. That's Awesome!

IMHO, `dbus` is the fundamental package for Redhat based OS(Centos, Fedora, openEuler...).
I'm afraid we may encounter other tricky issues in openEuler in the future if we don't have the dbus package in the future.

It would be appreciated that you could kindly put `dbus` installation into `bigtop/puppet` toolchain images.
Thank you very much.
@MacChen01

**Comment by MacChen01** [comment]
> > We found the reason that postgreSQL didn't start, becasue the dbus service didn't install and start in openEuler puppet default. So using the default Init postgreSQL mode, The code is: [cd6cd0d](https://github.com/apache/bigtop/commit/cd6cd0d62a4a648cdb6d2186af4f722a960d0ee3)
> > @guyuqi please check the code, thanks.
> 
> Thanks for your continuing efforts to improve this PR. That's Awesome!
> 
> IMHO, `dbus` is the fundamental package for Redhat based OS(Centos, Fedora, openEuler...). I'm afraid we may encounter other tricky issues in openEuler in the future if we don't have the dbus package in the future.
> 
> It would be appreciated that you could kindly put `dbus` installation into `bigtop/puppet` toolchain images. Thank you very much. @MacChen01

ok， We move `dbus` into `bigtop/puppet` toolchain, the code we modify in https://github.com/apache/bigtop/pull/1148
The commit is:
a7608972ca30309770468757b6e3cb547db1da41

@guyuqi pls check it, and we can verfiy the whole openEuler-support branch, thanks.

**Comment by uscxiexing** [comment]
> > > We found the reason that postgreSQL didn't start, becasue the dbus service didn't install and start in openEuler puppet default. So using the default Init postgreSQL mode, The code is: [cd6cd0d](https://github.com/apache/bigtop/commit/cd6cd0d62a4a648cdb6d2186af4f722a960d0ee3)
> > > @guyuqi please check the code, thanks.
> > 
> > 
> > Thanks for your continuing efforts to improve this PR. That's Awesome!
> > IMHO, `dbus` is the fundamental package for Redhat based OS(Centos, Fedora, openEuler...). I'm afraid we may encounter other tricky issues in openEuler in the future if we don't have the dbus package in the future.
> > It would be appreciated that you could kindly put `dbus` installation into `bigtop/puppet` toolchain images. Thank you very much. @MacChen01
> 
> ok， We move `dbus` into `bigtop/puppet` toolchain, the code we modify in #1148 The commit is: [a760897](https://github.com/apache/bigtop/commit/a7608972ca30309770468757b6e3cb547db1da41)
> 
> @guyuqi pls check it, and we can verfiy the whole openEuler-support branch, thanks.

The ambari smoketest log：
image
image
The detal log：
https://gitee.com/macchen1/bigtop/blob/master/smoke-test-log/aarch64/openeuler-22.03-smoke-log/ambari-openeuler-22.03.log

**Review by guyuqi — APPROVED** [review]

**Pull request merged**